src/main/java/org/jitsi/videobridge/octo/OctoPacket.java

```diff
@@ -0,0 +1,242 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.videobridge.octo;
+
+import org.jitsi.service.neomedia.*;
+import org.jitsi.util.*;
+
+/**
+ * A utility class which handles the on-the-wire Octo format. Octo encapsulates
+ * its payload (RTP, RTCP, or anything else) in an 8-byte header:
+ * <pre>{@code
+ *  0                   1                   2                   3
+ *  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |R| M | S | res |              Conference ID                    |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |                         Endpoint ID                           |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * }</pre>
+ * R: Source-is-a-relay flag. 1 if the source of the packet is a relay, and
+ * 0 if it is an endpoint (bridge).
+ * <p/>
+ * M: media type (audio, video, or data).
+ * <p/>
+ * S: Simulcast layer ID.
+ *
+ * @author Boris Grozev
+ */
+public class OctoPacket
+{
+    /**
+     * The fixed length of the Octo header.
+     */
+    public static final int OCTO_HEADER_LENGTH = 8;
+
+    /**
+     * The integer which identifies the "audio" media type in Octo.
+     */
+    public static final int OCTO_MEDIA_TYPE_AUDIO = 0;
+
+    /**
+     * The integer which identifies the "video" media type in Octo.
+     */
+    public static final int OCTO_MEDIA_TYPE_VIDEO = 1;
+
+    /**
+     * The integer which identifies the "data" media type in Octo.
+     */
+    public static final int OCTO_MEDIA_TYPE_DATA = 2;
+
+    /**
+     * @return the integer used to identify the particular {@link MediaType}
+     * in Octo.
+     */
+    private static int getMediaTypeId(MediaType mediaType)
+    {
+        switch (mediaType)
+        {
+        case AUDIO:
+            return OCTO_MEDIA_TYPE_AUDIO;
+        case VIDEO:
+            return OCTO_MEDIA_TYPE_VIDEO;
+        case DATA:
+            return OCTO_MEDIA_TYPE_DATA;
+        default:
+            return -1;
+        }
+    }
+
+    /**
+     * Writes an Octo header to the specified buffer at the specified offset.
+     * @param buf the buffer to write to.
+     * @param off the offset to write at.
+     * @param r the value of the {@code r} flag.
+     * @param mediaType the media type.
+     * @param s the value of the {@code s} flag.
+     * @param conferenceId the Octo ID of the conference.
+     * @param endpointId the Octo ID of the endpoint.
+     */
+    public static void writeHeaders(byte[] buf, int off,
```

Comment:
i thought in general we were favoring `RawPacket` (instead of buf, off, len) for these sorts of APIs?

Reply:
Yes, but Octo is implemented in the bridge only. It intercepts the DatagramPackets that libjitsi sends, so there isn't an existing RawPacket instance that I can use.